framework/Models/HybridModel.py

```diff
@@ -0,0 +1,621 @@
+    unknownList = utils.checkIfUnknowElementsinList(totalRomOutputs,modelOutputs)
+    if unknownList:
+      self.raiseAnError(IOError, "The following outputs: ", ','.join(str(e) for e in unknownList), " used in Model: ", modelName, "but not used in the paired ROMs.")
+
+  def getInitParams(self):
+    """
+      Method used to export to the printer in the base class the additional PERMANENT your local class have
+      @ In, None
+      @ Out, tempDict, dict, dictionary to be updated. {'attribute name':value}
+    """
+    tempDict = OrderedDict()
+    tempDict['ROMs contained in HybridModel are '] = self.romsDictionary.keys()
+    return tempDict
+
+  def getAdditionalInputEdits(self,inputInfo):
+    """
+      Collects additional edits for the sampler to use when creating a new input. In this case, it calls all the getAdditionalInputEdits methods
+      of the sub-models
+      @ In, inputInfo, dict, dictionary in which to add edits
+      @ Out, None.
+    """
+    self.modelInstance.getAdditionalInputEdits(inputInfo)
+
+  def __selectInputSubset(self,romName, kwargs):
+    """
+      Method aimed to select the input subset for a certain model
+      @ In, romName, string, the rom name
+      @ In, kwargs , dict, the kwarded dictionary where the sampled vars are stored
+      @ Out, selectedKwargs , dict, the subset of variables (in a swallow copy of the kwargs dict)
+    """
+    selectedKwargs = copy.copy(kwargs)
+    selectedKwargs['SampledVars'], selectedKwargs['SampledVarsPb'] = {}, {}
+    for key in kwargs["SampledVars"].keys():
+      if key in self.romsDictionary[romName]['Instance'].getInitParams()['Features']:
+        selectedKwargs['SampledVars'][key] = kwargs["SampledVars"][key]
+        selectedKwargs['SampledVarsPb'][key] = kwargs["SampledVarsPb"][key] if 'SampledVarsPb' in kwargs.keys() else 1.0
+    return selectedKwargs
+
+  def createNewInput(self,myInput,samplerType,**kwargs):
+    """
+      This function will return a new input to be submitted to the model, it is called by the sampler.
+      @ In, myInput, list, the inputs (list) to start from to generate the new one
+      @ In, samplerType, string, is the type of sampler that is calling to generate a new input
+      @ In, **kwargs, dict,  is a dictionary that contains the information coming from the sampler,
+           a mandatory key is the sampledVars'that contains a dictionary {'name variable':value}
+      @ Out, newInputs, dict, dict that returns the new inputs for each sub-model
+    """
+    self.raiseADebug("Create New Input")
+    if self.romValid:
+      identifier = kwargs['prefix']
+      newKwargs = {'prefix':identifier}
+      for romName in self.romsDictionary.keys():
+        newKwargs[romName] = self.__selectInputSubset(romName, kwargs)
+        newKwargs[romName]['prefix'] = romName+utils.returnIdSeparator()+identifier
+        newKwargs[romName]['uniqueHandler'] = self.name+identifier
+    else:
+      newKwargs = copy.deepcopy(kwargs)
```

Comment:
remember to store the real "identifier" somewhere and to push it back in the metadata since there are some Samplers and Optimizers that use the identifiers (prefix) to search for particular data (e.g. SPSA for identify the perturbation for the gradient evaluations)

Reply:
I not sure for this, but I do reset the prefix for each run.